native/src/boot/magiskboot.hpp

```diff
@@ -4,19 +4,21 @@
 
 #include <base.hpp>
 
-#define HEADER_FILE         "header"
-#define KERNEL_FILE         "kernel"
-#define RAMDISK_FILE        "ramdisk.cpio"
-#define SECOND_FILE         "second"
-#define EXTRA_FILE          "extra"
-#define KER_DTB_FILE        "kernel_dtb"
-#define RECV_DTBO_FILE      "recovery_dtbo"
-#define DTB_FILE            "dtb"
-#define VENDOR_RAMDISK_FILE "vendor_ramdisk_%.*s.cpio"
-#define BOOTCONFIG_FILE     "bootconfig"
-#define NEW_BOOT            "new-boot.img"
+#define HEADER_FILE     "header"
+#define KERNEL_FILE     "kernel"
+#define RAMDISK_FILE    "ramdisk.cpio"
+#define SECOND_FILE     "second"
+#define EXTRA_FILE      "extra"
+#define KER_DTB_FILE    "kernel_dtb"
+#define RECV_DTBO_FILE  "recovery_dtbo"
+#define DTB_FILE        "dtb"
+#define NEW_BOOT        "new-boot.img"
+#define BOOTCONFIG_FILE "bootconfig"
+#define VENDOR_RAMDISK_FILE_PREFIX  "vendor_ramdisk_"
+#define VENDOR_RAMDISK_FILE_SUFFIX  ".cpio"
+#define VENDOR_RAMDISK_FILE         VENDOR_RAMDISK_FILE_PREFIX "%.*s" VENDOR_RAMDISK_FILE_SUFFIX
```

Comment:
since the length is always `VENDOR_RAMDISK_NAME_SIZE`, include `flags.h` and

```suggestion
#define VENDOR_RAMDISK_FILE VENDOR_RAMDISK_FILE_PREFIX "%." str(VENDOR_RAMDISK_NAME_SIZE) "s" VENDOR_RAMDISK_FILE_SUFFIX
```

Reply:
If we are gonna support remove/add entry, maybe we don't need this, and instead we do something like:

### dump
```c++
ssprintf(filename, VENDOR_RAMDISK_FILE_PREFIX "%." str(VENDOR_RAMDISK_NAME_SIZE) "s" VENDOR_RAMDISK_FILE_SUFFIX, entry.ramdisk_name);
```

### repack & cleanup
```c++
vendor_ramdisk_table_entry_v4 tmp;
while (struct dirent *dir = xreaddir(d.get())) {
   if (sscanf(dir->d_name, VENDOR_RAMDISK_FILE_PREFIX "%" str(VENDOR_RAMDISK_NAME_SIZE) "s" VENDOR_RAMDISK_FILE_SUFFIX, tmp.ramdisk_name) == 1) {
       // read to tmp or unlink dir->d_name
   }
}
```

Reply:
done